neuralpredictors/layers/readouts/multi_readout.py

```diff
@@ -1,107 +1,147 @@
+        return self[data_key].regularizer(reduction=reduction, average=average)
 
-    # Make sure this is not a bug
-    def regularizer(self, readout_key):
-        return self.gamma_readout
 
-
-class MultiplePointPooled2d(MultiReadout):
+class MultiReadoutSharedParametersBase(MultiReadoutBase):
     """
-    Instantiates multiple instances of PointPool2d Readouts
-    usually used when dealing with more than one dataset sharing the same core.
+    Base class for MultiReadouts that share parameters between readouts.
+    For more information on which parameters can be shared, refer for example to the FullGaussian2d readout
     """
 
+    def prepare_readout_kwargs(self, i, data_key, first_data_key, **kwargs):
+        readout_kwargs = kwargs.copy()
+
+        if "grid_mean_predictor" in readout_kwargs:
+            if readout_kwargs["grid_mean_predictor"] is not None:
+                if readout_kwargs["grid_mean_predictor_type"] == "cortex":
+                    readout_kwargs["source_grid"] = readout_kwargs["source_grids"][data_key]
+                else:
+                    raise KeyError(
+                        "grid mean predictor {} does not exist".format(readout_kwargs["grid_mean_predictor_type"])
+                    )
+                if readout_kwargs["share_transform"]:
+                    readout_kwargs["shared_transform"] = None if i == 0 else self[first_data_key].mu_transform
+
+            elif readout_kwargs["share_grid"]:
+                readout_kwargs["shared_grid"] = {
+                    "match_ids": readout_kwargs["shared_match_ids"][data_key],
+                    "shared_grid": None if i == 0 else self[first_data_key].shared_grid,
+                }
+
+            del readout_kwargs["share_transform"]
+            del readout_kwargs["share_grid"]
+            del readout_kwargs["grid_mean_predictor_type"]
+
+        if "share_features" in readout_kwargs:
+            if readout_kwargs["share_features"]:
+                readout_kwargs["shared_features"] = {
+                    "match_ids": readout_kwargs["shared_match_ids"][data_key],
+                    "shared_features": None if i == 0 else self[first_data_key].shared_features,
+                }
+            else:
+                readout_kwargs["shared_features"] = None
+            del readout_kwargs["share_features"]
+        return readout_kwargs
+
+
+#### MultiReadouts for backwards compatibility
+class MultiplePointPooled2d(MultiReadoutBase):
     _base_readout = PointPooled2d
 
 
-class MultipleFullGaussian2d(MultiReadout):
-    """
-    Instantiates multiple instances of FullGaussian2d Readouts
-    usually used when dealing with more than one dataset sharing the same core.
-
-    Args:
-        in_shape (list): shape of the input feature map [channels, width, height]
-        loaders (list):  a list of dataloaders
-        gamma_readout (float): regularizer for the readout
-    """
+class MultipleSpatialXFeatureLinear(MultiReadoutBase):
```

Comment:
Could you also add a `MultipleFactorized2d` here? (or in the __init__, see comment above), then we have the important readouts already ready to import an use.